lib/migration.js

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (c) 2012-2016 Digital Bazaar, Inc. All rights reserved.
+ */
+
+var _ = require('lodash');
+var async = require('async');
+var bedrock = require('bedrock');
+var config = bedrock.config;
+var database = require('./database');
+var progress = require('progress');
+
+var BedrockError = bedrock.util.BedrockError;
+
+var logger = bedrock.loggers.get('app');
+
+var api = {};
+module.exports = api;
+
+// steps per progress bar tick call optimization
+var PROCESSED_PER_TICK = 20;
+
+bedrock.events.on('bedrock-cli.init', function() {
+  // setup common migration options
+  bedrock.program
+    .option('--print-objects', 'Print out changed objects.')
+    .option('--progress', 'Print progress.');
+});
+
```

Comment:
If this tool should require the `--config` option to setup database config and such then something like the following might work:
```js
bedrock.events.on('bedrock-cli.ready', function() {
  if(bedrock.program.config.length === 0) {
    throw new Error('Database setup via the --config option is required.');
  }
});
```
The above doesn't force you to setup the database, but would be a good hint. The config values could be checked here too but I'm not sure how to distinguish between the default values and use specified values.